client/me/purchases/purchase-item/index.jsx

```diff
@@ -55,65 +49,93 @@ class PurchaseItem extends Component {
 		);
 	}
 
-	renewsOrExpiresOn() {
+	getStatus() {
 		const { purchase, translate, moment } = this.props;
+		const expiry = moment( purchase.expiryDate );
 
-		if ( showCreditCardExpiringWarning( purchase ) ) {
-			return (
-				<Notice isCompact status="is-error" icon="notice">
-					{ translate( 'Credit card expiring soon' ) }
-					{ this.trackImpression( 'credit-card-expiring' ) }
-				</Notice>
-			);
+		if ( purchase && isPartnerPurchase( purchase ) ) {
+			return translate( 'Managed by %(partnerName)s', {
+				args: {
+					partnerName: getPartnerName( purchase ),
+				},
+			} );
 		}
 
 		if ( isRenewing( purchase ) && purchase.renewDate ) {
 			const renewDate = moment( purchase.renewDate );
 
-			return translate( 'Renews at %(amount)s on %(date)s.', {
+			if ( creditCardHasAlreadyExpired( purchase ) ) {
+				return (
+					<span className="purchase-item__is-error">
+						{ translate( 'Credit card expired' ) }
+						{ this.trackImpression( 'credit-card-expiring' ) }
+					</span>
+				);
+			}
+
+			if ( creditCardExpiresBeforeSubscription( purchase ) ) {
+				return (
+					<span className="purchase-item__is-error">
+						{ translate( 'Credit card expiring soon' ) }
+						{ this.trackImpression( 'credit-card-expiring' ) }
+					</span>
+				);
+			}
+
+			return translate( 'Renews at %(amount)s on {{span}}%(date)s{{/span}}', {
 				args: {
 					amount: purchase.priceText,
 					date: renewDate.format( 'LL' ),
 				},
+				components: {
+					span: <span className="purchase-item__date" />,
+				},
 			} );
 		}
 
-		const expiry = moment( purchase.expiryDate );
-
 		if ( isExpiring( purchase ) ) {
-			if ( expiry < moment().add( 30, 'days' ) ) {
-				const status = isRecentMonthlyPurchase( purchase ) ? 'is-info' : 'is-error';
+			if ( expiry < moment().add( 30, 'days' ) && ! isRecentMonthlyPurchase( purchase ) ) {
```

Comment:
I'm not sure this is a good change -- for example, if I have a monthly subscription with auto-renew off that expires tomorrow, shouldn't I always see a message about that?

I think it also makes things inconsistent with what you see when you click through to the subscription (I'm pretty sure we do show a message highlighting the upcoming expiration there).

Reply:
The `isRecentMonthlyPurchase` checks whether the purchase was made in the last week so if you bought a monthly plan, turned auto-renew off, and your purchase expires tomorrow, then this statement will be true.

Reply:
Good point -- somehow I missed that `isRecentMonthlyPurchase` was only true for the first week.  This seems OK.